.github/workflows/pytests.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Server PyTests'
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - develop
+  push:
+    branches:
+      - develop
+    paths:
+      - "deploy/**"
+  workflow_dispatch:
+jobs:
+  pytests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.2'
```

Comment:
please use `3.12.8` which is a actual version of 3.12 track, or just define `3.12` to use the latest 3.12.x

ref: https://www.python.org/downloads/release/python-3128/